vm/src/frame.rs

```diff
@@ -393,9 +393,8 @@ impl Frame {
                         // Take all key-value pairs from the dict:
                         let dict: PyDictRef =
                             obj.downcast().expect("Need a dictionary to build a map.");
-                        let dict_elements = dict.get_key_value_pairs();
-                        for (key, value) in dict_elements.iter() {
-                            map_obj.set_item(key.clone(), value.clone(), vm).unwrap();
+                        for (key, value) in dict {
```

Comment:
This is awesome, good improvement!

Reply:
It's fun, isn't. It's lovely how simple things become if you get right traits in place.